### DISCUSSION TITLE
Using Budibase API through REST Connector to Grant Access (Add User)

### DISCUSSION THREAD

**Discussion by JayP718** [Budibase/budibase#8572] [Help]
Hi,

I'm trying to grant access to budibase through the api using Budibase's rest api connector. I've validated that my requests is right on my local machine but I'm having trouble sending this requests through the rest api connector.

I keep getting 
{
  "message": "No user info found",
  "status": 403
}

Please see headers and body below.

![image](https://user-images.githubusercontent.com/42209052/200418962-84dce21f-8fe8-409b-a6aa-8f9b0493c27e.png)

![image](https://user-images.githubusercontent.com/42209052/200419199-5e1746f3-fa20-4a47-8d9b-58433df948c4.png)

Any help is appreciated!

**Comment by melohagan** [comment]
Hey @JayP718 

Could you show me which endpoint you are trying to hit please?

**Reply by JayP718** [reply]
Hi Mel,

It's self hosted but I just use the IP address of the server so http://[MYIPADDRESSHERE]:10000/api/public/v1/users. I do get a response so I think the endpoint is right but I'm not positive.

**Reply by melohagan** [reply]
Thank you clarifying @JayP718 

I believe the error message is a bit misleading and that the main issue is the 403 error code - i.e. unauthorised.

I was able to replicate the error message by changing a character in the API key. Double check that your API Key is definitely correct!

**Reply by JayP718** [reply]
HI @melohagan,

So I copied it from the api exactly and I'm still getting the same error :(. My api key does have a "-" character so i'm unsure if that has anything to do with it.

Am I right in putting that JSON in the body of the post request?
Do my headers look right?

Thanks for having a look

**Reply by melohagan** [reply]
That is strange, I cannot recreate this issue. Is the active toggle on for the api key header?

**Reply by JayP718** [reply]
Yes, I'm having this issue on the self hosted version of budibase but I'm trying to test on the hosted version and I'm also having issues.

Not sure if you are able to see it on your end but here's the end point.

https://a3datasolutionsllc.budibase.app/

**Reply by JayP718** [reply]
It's in the School Admin App

**Reply by JayP718** [reply]
Hi Mel,

Just following up on this as I feel it's a very simple oversight holding me back since your unable to recreate . Is there anything different in my app from yours?

**Reply by melohagan** [reply]
Hey @JayP718 I am unable to access the app via that link, would you be able to email the app export to support@budibase.com?

**Reply by JayP718** [reply]
@melohagan Thank you for replying so quickly, I've emailed you the app.

Jay

**Reply by melohagan** [reply]
Hey @JayP718 I don't see an email in my inbox. Would you mind resending please?

**Reply by JayP718** [reply]
> Hey @JayP718 I don't see an email in my inbox. Would you mind resending please?

Hi @melohagan ,

Just resent from a different email. Let me know if you see it please.

**Reply by melohagan** [reply]
Hmm... still no email. You're definitely sending it to: support@budibase.com ?

**Reply by JayP718** [reply]
For security reasons, Gmail does not allow you to use this type of file as it violates Google policy for executables and archives.


Getting this error in Gmail 


I'll try Yahoo

**Reply by melohagan** [reply]
If you're still having issues @JayP718 then feel free to drop me a message on Discord. Hopefully you can send it through there.
